### PR TITLE
added license to docs and updated spec_v3

### LINKED ISSUE #1180: Add License to [Zarr documentation](https://zarr.readthedocs.io/)
I think it would be good to have the [LICENSE](https://github.com/zarr-developers/zarr-python/blob/main/LICENSE) added as a page on the [documentation](https://zarr.readthedocs.io/). So it can appear on the side panel.

### LINKED ISSUE #1181: Duplicate label spec_v2 in /docs/spec/v3.rst and /docs/spec/v2.rst
Command
```
tox -e docs
```
ends in an error.
 
I think the label in /docs/spec/v3.rst should be 
```
.. _spec_v3:
```
and not 
```
.. _spec_v2:
```

### PR DESCRIPTION
**Pull request by DON-BRAN**
fixes #1180
fixes #1181

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

File: docs/license.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
+License
+=======
+
+The MIT License (MIT)
```
Comment (jakirkham) [inline docs/license.rst:4]: Is there some way to include [this license file]( https://github.com/zarr-developers/zarr-python/blob/main/LICENSE ) without copying it?

File: docs/spec/v3.rst [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.. _spec_v2:
+.. _spec_v3:
```
Comment (jakirkham) [inline docs/spec/v3.rst:1]: This change looks ok to me

cc @joshmoore (in case you have thoughts)

**Review by jakirkham — COMMENTED** [review]
Thanks! 🙏

Had a couple comments below

File: docs/license.rst (hunk 1 above)
Comment (jakirkham) [reply docs/license.rst]: Ah sorry, let me clarify. Is there a way with Sphinx to tell it to load the content from `LICENSE` as opposed to creating a second copy of it here? IOW would it be possible to drop this `rst` file somehow?

Comment (jakirkham) [reply docs/license.rst]: No worries. Can see how that might be unclear.

Maybe this [SO question is helpful]( https://stackoverflow.com/q/2921724 )? We could also add a symlink if being outside `docs` is an issue

Comment (DON-BRAN) [reply docs/license.rst]: from https://www.sphinx-doc.org/en/master/usage/markdown.html
I think `myst-parser`has to be installed as well

Comment (DON-BRAN) [reply docs/license.rst]: Good morning @jakirkham. I wasn't able to completely remove the license.rst file but i could link it to the original LICENSE file without copying the content.
I referenced this issue https://github.com/sphinx-doc/sphinx/issues/7000

Comment (jakirkham) [reply docs/license.rst]: Actually this is exactly what I was looking for. Thank you! 😄

File: docs/spec/v3.rst (hunk 2 above)
Comment (jakirkham) [reply docs/spec/v3.rst:1]: Sounds like this is good

xref: https://github.com/zarr-developers/zarr-python/issues/1181#issuecomment-1277444650

**Review by jakirkham — APPROVED** [review]
LGTM. Once CI passes this should be in. Thanks @DON-BRAN! 🙏

**Pull request merged**